sdk/lib/rtl/heap.c

```diff
@@ -353,21 +353,23 @@ RtlpInsertFreeBlockHelper(PHEAP Heap,
     Heap->FreeHints[HintIndex] = &FreeEntry->FreeList;
 }
 
-VOID NTAPI
+LONG NTAPI
 RtlpInsertFreeBlock(PHEAP Heap,
                     PHEAP_FREE_ENTRY FreeEntry,
                     SIZE_T BlockSize)
 {
     USHORT Size, PreviousSize;
     UCHAR SegmentOffset, Flags;
     PHEAP_SEGMENT Segment;
+    LONG FreeSize;
 
     DPRINT("RtlpInsertFreeBlock(%p %p %x)\n", Heap, FreeEntry, BlockSize);
 
     /* Increase the free size counter */
     Heap->TotalFreeSize += BlockSize;
 
     /* Remember certain values */
+    FreeSize = FreeEntry->Size;
```

Comment:
Note that `FreeEntry->Size` is still a USHORT variable...

Reply:
This is fine. See comment above. Thanks.

Reply:
FreeSize variable has now been removed. Thanks.